Fix: Error when deleting local changes.

Konrad-Simso opened the pull request:
## Description

This PR includes the following changes:
- Bugfix
- Refactoring of a query to split into a mutation and a useQuery.
- Moved a common function between 2 files to be in a utils folder.
- Added a test for new utils function and update old tests.

The bugfix was related to the order of when certain components were rerendering. This has been fixed by changing what is invalidated in the cache during deletion and when to load certain components based on query results.

## Related Issue(s)
- #13494 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)

mlqn reviewed (commented):
Nice work! 🚀 I just have one suggestion regarding the validation of the `selectedOption` / `modelPath`

Konrad-Simso commented:
> Nice work! 🚀 I just have one suggestion regarding the validation of the `selectedOption` / `modelPath`

Your suggestions makes the bug exist again. 
Without filtering in both places, SchemaEditorWithToolbar.tsx and TopToolbar.tsx, then some component will rerender with the wrong modelPath selected. I've added the filter to both files and removed the Utils function in the last commit.

Is this an okay solution?

mlqn commented:
I don't think we need to duplicate the find in both components. Did you pass the `existingSelectedOption` to the `TopToolbar` when you tested?
`selectedOption={existingSelectedOption}`

Konrad-Simso commented:
> I don't think we need to duplicate the find in both components. Did you pass the `existingSelectedOption` to the `TopToolbar` when you tested? `selectedOption={existingSelectedOption}`

i'll try it

JamalAlabdullah reviewed (approved):
Tested ok 👍
